### PR TITLE
[ci] fixed r-lintr version constraint

### LINKED ISSUE #2986: [ci] R linter is broken
We have the following error in `master`:
```
Linting R code
Error: 'absolute_path_linter' is not an exported object from 'namespace:lintr'
Execution halted
```

https://travis-ci.org/github/microsoft/LightGBM/jobs/672985225

### PR DESCRIPTION
**Pull request by StrikerRUS**
Fixed #2986.

### CODE REVIEW

**Comment by StrikerRUS** [comment]
> This is WILD how did you know to try this?

It was the first thing that came to my mind when I saw your words about 

 > I see the first conda install reflected in the logs
  but not the second one with libxml2 and r-lintr

**Comment by jameslamb** [comment]
> > This is WILD how did you know to try this?
> 
> It was the first thing that came to my mind when I saw your words about
> 
> > I see the first conda install reflected in the logs
> > but not the second one with libxml2 and r-lintr

amazing

**Comment by StrikerRUS** [comment]
> Also I see this issue on Mac R builds now 🤒

```
curl: (22) The requested URL returned error: 404 Not Found
```

**Comment by jameslamb** [comment]
> > Also I see this issue on Mac R builds now 🤒
> 
> ```
> curl: (22) The requested URL returned error: 404 Not Found
> ```

It has to be a transient thing, right? Because the Mac builds on Travis use the exact same code and worked fine

**Comment by StrikerRUS** [comment]
> It has to be a transient thing, right?

I hope so... Or it's something wrong with certificates like in #2977.
Let's wait a little bit a see 🙂

**Comment by StrikerRUS** [comment]
I guess here is the root cause:
https://github.com/Homebrew/homebrew-cask/pull/80370

> May take time for all mirrors around the world to update.

**Comment by StrikerRUS** [comment]
> Let's wait a little bit a see

OK, I think it's time to ask what's going on: https://github.com/Homebrew/homebrew-cask/pull/80370#issuecomment-612246863.

**Comment by StrikerRUS** [comment]
Hmm, seems that 24h is not enough to update the URL. Guess we should wait till Monday.

**Comment by StrikerRUS** [comment]
Still no updates for some other mirros (they are changed automatically when I re-run CI job).

File: .ci/test_r_package.sh [hunk 1]
```diff
@@ -35,6 +35,10 @@ fi
 # Installing R precompiled for Mac OS 10.11 or higher
 if [[ $OS_NAME == "macos" ]]; then
 
+    # temp fix for basictex
+    if [[ $AZURE == "true" ]]; then
+        brew update
+    fi
```
Comment (jameslamb) [inline .ci/test_r_package.sh:41]: Why is this change only for AZURE? just to save build time?

ugh the Azure Mac builds are such a pain. I had to do another "if Azure" thing on a PR I'm working on: 

image

Comment (StrikerRUS) [reply .ci/test_r_package.sh:41]: > Why is this change only for AZURE? just to save build time?

Yeah! Seems that Travis does it automatically during `setup.sh`.

Comment (jameslamb) [reply .ci/test_r_package.sh:41]: I saw those exact warnings actually. Thanks for the reminder!

Still just experimenting with things 😬

**Comment by jameslamb** [comment]
hey glad that it worked! Thank you!!

**Pull request merged**